app/screens/packageDetails/packageDetails.tsx

```diff
@@ -1,70 +1,154 @@
-import * as React from "react"
-import { SafeAreaView, StyleSheet, View } from "react-native"
-import { Button, Screen, Text } from "../../components"
-import { PackageData, PackageStatus } from "../packagesList/types"
+import React, { FC, ReactElement, useMemo } from "react"
+import { SafeAreaView, StyleSheet, View, Linking, TouchableOpacity } from "react-native"
+import { Button, Header, Icon, Screen, Text } from "../../components"
+import { PackageData, PackageStatus, PackageStatusAPI } from "../packagesList/types"
 import { PackageStatusHeader } from "./packageStatusHeader"
 import { NavigationInjectedProps } from "react-navigation"
 import { color } from "../../theme"
+import { useStores } from "../../models/root-store"
+import { observer } from "mobx-react-lite"
+import { IconTypes } from "../../components/icon/icons"
+import { IS_IOS } from "../../constants/constants"
 
 interface PackageDetailsScreenProps {
     packageData: PackageData
 }
 
-export const PackageDetailsScreen: React.FunctionComponent<NavigationInjectedProps<PackageDetailsScreenProps>> = props => {
+export const PackageDetailsScreen: FC<NavigationInjectedProps<PackageDetailsScreenProps>> = observer(props => {
   const packageData = props.navigation.state.params.packageData
-  const renderFullNameView = (): React.ReactElement => {
+  const goBack = useMemo(() => () => props.navigation.goBack(null), [props.navigation])
+
+  const { packagesStore: { updatePackagesStatus } } = useStores()
+
+  const onIconPressed = (iconName: IconTypes): void => {
+    const { phoneNumber, destination: { city, street, number } } = packageData
+    const formattedAddress = `${street} ${number} ${city}`
+    const nativeNavigationAppURL = IS_IOS
+      ? `https://maps.apple.com/?daddr=${formattedAddress}`
+      : `https://www.google.com/maps/dir/?api=1&travelmode=driving&dir_action=navigate&destination=${formattedAddress}`
+
+    switch (iconName) {
+      case "call": { Linking.openURL(`tel:${phoneNumber}`); break }
+      case "msg": { Linking.openURL(`sms:${phoneNumber}`); break }
+      case "waze": { Linking.openURL(`https://waze.com/ul?q=${formattedAddress}`); break }
+      case "location": { Linking.openURL(nativeNavigationAppURL); break }
+      default: break
+    }
+  }
+
```

Comment:
logic and UI of the icon should be gathered inside a separate component - linkingIcon or so...